core/jobs/batch_jobs/topic_validation_jobs.py

```diff
@@ -0,0 +1,95 @@
+
+# coding: utf-8
+#
+# Copyright 2022 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Validation Jobs for topic"""
+
+from __future__ import annotations
+
+from core.domain import topic_fetchers
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+
+(topic_models, ) = models.Registry.import_models([models.NAMES.topic])
+
+
+class GetNumberOfTopicsWhereStoryIsPublishedNotBoolJob(base_jobs.JobBase):
+    """Job that returns topics where story_is_published value is not bool."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns PCollection of invalid topics which have story_is_published
+        value is not bool with their id.
+
+        Returns:
+            PCollection. Returns PCollection of invalid topics which have
+            story_is_published value is not bool with their id.
+        """
+        total_topics = (
+            self.pipeline
+            | 'Get all TopicModels' >> ndb_io.GetModels(
+                topic_models.TopicModel.get_all(include_deleted=False))
+            | 'Get topic from model' >> beam.Map(
+                topic_fetchers.get_topic_from_model)
+        )
+
+        def checking_bool_values(references):
+            for reference in references:
+                if not isinstance(reference.story_is_published, bool):
+                    return True
+            return False
```

Comment:
Move this out of this function into the class.

Reply:
Done